### PR TITLE
Configure project's `languageVersion` via toolchain

### PR DESCRIPTION
**Pull request by Goooler**
https://kotlinlang.org/docs/gradle-configure-project.html#gradle-java-toolchains-support

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -85,12 +85,6 @@ allprojects {
         }
     }
 
-    tasks.withType(KotlinCompile) {
-        kotlinOptions {
-            jvmTarget = libs.versions.java.get()
-        }
```
Comment (Goooler) [inline build.gradle:91]: Deprecated from Kotlin 1.8.0

**Comment by cketti** [comment]
As far as I can tell we don't want `sourceCompatibility` and `targetCompatibility` to be linked to the toolchain language version.

**Comment by Goooler** [comment]
https://issuetracker.google.com/issues/260059413#comment2

**Comment by cketti** [comment]
We don't care much which JDK version is used to compile the app (as long as it is new enough). Because we're building an Android project, we do care about the target version and hence the source version. We already explicitly set these to 11, so there's no need to change anything.

**Comment by Goooler** [comment]
The motivation is to replace the deprecated `kotlinOptions.jvmTarget`, if we use the recommended way of toolchain to do that, will unify all of `sourceCompatibility` & `targetCompatibility`, I believe that is a proper solution.

**Comment by cketti** [comment]
I've created #6603 to replace `kotlinOptions.jvmTarget`.

**Pull request closed without merging**